core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/UniqueIDReadWriteLDAPUserStoreManager.java

```diff
@@ -1731,15 +1735,18 @@ private String resolveGroupDN(String searchFilter, String role, LDAPRoleContext
         }
     }
 
-    private String getGroupName(SearchResult resultedGroup) throws NamingException {
+    private String getGroupName(SearchResult resultedGroup, String searchBase) throws NamingException {
 
         Attribute attribute = resultedGroup.getAttributes()
                 .get(realmConfig.getUserStoreProperty(LDAPConstants.GROUP_NAME_ATTRIBUTE));
         if (attribute == null) {
             return resultedGroup.getName();
         } else {
-            String groupNameAttributeValue = (String) attribute.get();
-            return realmConfig.getUserStoreProperty(LDAPConstants.GROUP_NAME_ATTRIBUTE) + "=" + groupNameAttributeValue;
+            String groupDN = StringUtils.replace(resultedGroup.getName(), searchBase, EMPTY_ATTRIBUTE_STRING);
+            if (StringUtils.isNotBlank(groupDN) && StringUtils.endsWith(groupDN, ",")) {
```

Comment:
Can you a reusable private method for this modification as the logic is repeated twice.

Reply:
This method is private, are the same method is using two files.